multicluster/service-mirror/cluster_watcher.go

```diff
@@ -490,8 +461,14 @@ func (rcsw *RemoteClusterServiceWatcher) handleRemoteServiceCreated(ctx context.
 	serviceInfo := fmt.Sprintf("%s/%s", remoteService.Namespace, remoteService.Name)
 	localServiceName := rcsw.mirroredResourceName(remoteService.Name)
 
-	if err := rcsw.mirrorNamespaceIfNecessary(ctx, remoteService.Namespace); err != nil {
-		return err
+	// Ensure the namespace exists, and skip mirroring if it doesn't
+	if _, err := rcsw.localAPIClient.NS().Lister().Get(remoteService.Namespace); err != nil {
+		if kerrors.IsNotFound(err) {
+			rcsw.log.Warnf("Skipping mirroring of service %s: namespace %s does not exist", serviceInfo, remoteService.Namespace)
```

Comment:
Should we also emit a [`ServiceMirroringSkipped`](https://github.com/linkerd/linkerd2/blob/main/multicluster/service-mirror/cluster_watcher.go#L28) event, or would that complicate the code without adding too much value?

Reply:
I agree I think we should add a `rcsw.recorder.Event` here similar to when we [skip mirroring](https://github.com/linkerd/linkerd2/blob/9ad003c523394fd47fc8b2b0890cc0f1185b2e07/multicluster/service-mirror/cluster_watcher_headless.go#L50) headless services.